### DISCUSSION TITLE
How do you generate, import, export or template dashboard

### DISCUSSION THREAD

**Discussion by rangzen** [uptrace/uptrace#165] [Q&A] [answered]
Hello, I am doing a lot of tests actually and creating, deleting, changing, and testing dashboards is cumbersome. Can I generate a .yaml file and import it? Do I use uptrace.yml directly? Do I have to inject it directly into Uptrace's database? Thank you.

**Comment by vmihailenco** [comment, marked as answer]
Hey,

I am generating templates by directly editing [YAML](https://github.com/uptrace/uptrace/tree/master/config/dashboard-templates) files and [recompiling](https://uptrace.dev/get/contributing.html#compiling-uptrace-collector) Uptrace. You can drop your own files in that directory and Uptrace will pick them up.

We plan on providing some UI for exporting and importing dashboards via YAML files, but that have to wait until we support [tables and heatmaps](https://github.com/uptrace/cloud-changelog/issues/11).

Here is what I am working on and it will be soon available for testing at Uptrace.dev

![metrics-table-heatmap](https://user-images.githubusercontent.com/290976/220045596-710051b7-d46e-4f7e-b551-49b407e064e6.png)

I hope to finish that work by March and start porting the changes to this repo.

>and testing dashboards is cumbersome

I am working on metrics UI right now so any feedback you might have is more than welcome.

**Reply by rangzen** [reply]
I am working on https://github.com/rangzen/otel-status and you can check the screenshot in `/docs` because I am using Uptrace for testing right now.
For example, how do you rename a grouping part? Or extract a part of the naming? How get `http://localhost:8082 (2xx)` from `status{http.status_class=2xx,http.url=http://localhost:8082}`?
You said you work on color choice which is perfect :+1: 
A great shortcut will be to be able to save the result when you play with Explore. You try this and this, you find the perfect configuration and ... you have to remember everything and do it again in a dashboard.
Thank you for your work!

**Reply by vmihailenco** [reply]
>How get http://localhost:8082 (2xx) from status{http.status_class=2xx,http.url=http://localhost:8082}?

This could be an option to specify template like `${http.url} (${http.status_class})`. I've created https://github.com/uptrace/uptrace/issues/175 to track this.

>You said you work on color choice which is perfect +1

Yes, this is almost done.

>you find the perfect configuration and ... you have to remember everything and do it again in a dashboard.

:+1: I don't see an easy solution right, but I acknowledge the problem.

**Comment by rangzen** [comment]
Is the sorting random? By dates?
![image](https://user-images.githubusercontent.com/132700/221368030-f61dc986-4b42-489e-85a2-7166bd393c95.png)

**Reply by vmihailenco** [reply]
It is random / no sorting. We probably should sort by the name. Created https://github.com/uptrace/uptrace/issues/176

**Comment by rangzen** [comment]
Can I exclude 0 values from the display?
![image](https://user-images.githubusercontent.com/132700/221368097-efa0b200-e560-4eee-8187-33b8433bbc21.png)

**Reply by vmihailenco** [reply]
Hard to say, but probably yes. It is probably should be done via the query language, but we don't support that yet.

But I've already made the tooltip smaller and added a pagination to the timeseries.
